File: src/control/actor.py
import torch

N, D_o, H1, H2, H3, D_a = 1, 6*31, 100, 100, 100, 6

o = torch.randn(D_o)

model = torch.nn.Sequential(
  torch.nn.Linear(D_o, H1),
  torch.nn.ReLU(),
  torch.nn.Linear(H1, H2),
  torch.nn.ReLU(),
  torch.nn.Linear(H2, H3),
  torch.nn.ReLU(),
  torch.nn.Linear(H3, D_a)
)


traced = torch.jit.trace(model, o)
traced.save('actor.pt')

File: src/control/actor.cpp
//
// Created by amiani on 4/20/20.
//

#include "actor.h"

Actor::Actor(::String path) : module(torch::jit::load(path)) {
  module.to(DEVICE);
}

Actor::Actor(torch::jit::script::Module m) : module(m) {
  module.to(DEVICE);
}
